protostar/commands/invoke_command.py

```diff
@@ -0,0 +1,145 @@
+                description="The name of the function being called.",
+                type="felt",
+                is_required=True,
+                is_array=True,
+            ),
+            ProtostarArgument(
+                name="max-fee",
+                description=(
+                    "The maximum fee that the sender is willing to pay for the transaction. "
+                    'Provide "auto" to auto estimate the fee.'
+                ),
+                type="fee",
+            ),
+            ProtostarArgument(
+                name="wait-for-acceptance",
+                description="Waits for transaction to be accepted on chain.",
+                type="bool",
+                default=False,
+            ),
+        ]
+
+    async def run(self, args: Any):
+        network_command_util = NetworkCommandUtil(args, self._logger)
+        signable_command_util = SignableCommandUtil(args, self._logger)
+        network_config = network_command_util.get_network_config()
+        gateway_client = network_command_util.get_gateway_client()
+        signer = signable_command_util.get_signer(network_config)
+
+        return await self.invoke(
+            address=args.contract_address,
+            function=args.function,
+            inputs=args.inputs,
+            gateway_client=gateway_client,
+            signer=signer,
+            max_fee=args.max_fee,
+            wait_for_acceptance=args.wait_for_acceptance,
+            account_address=args.account_address,
+        )
+
+    async def invoke(
+        self,
+        address: int,
+        function: str,
+        inputs: list[int],
+        gateway_client: GatewayClient,
+        signer: Optional[BaseSigner] = None,
+        account_address: Optional[str] = None,
+        max_fee: Optional[Fee] = None,
+        wait_for_acceptance: bool = False,
+    ):
+        gateway_facade = self._gateway_facade_factory.create(
+            gateway_client=gateway_client, logger=None
+        )
+        if account_address is None:
+            raise ProtostarException(
+                "Argument `account_address` is required for transactions V1."
+            )
+        if max_fee is None:
+            raise ProtostarException(
+                "Argument `max-fee` is required for transactions V1."
+            )
+        if signer is None:
+            raise ProtostarException(
+                "Argument `signer` is required for transactions V1 when private-key is not detected."
+            )
+
+        response = await gateway_facade.invoke(
+            contract_address=address,
+            function_name=function,
+            inputs=inputs,
+            max_fee=max_fee if isinstance(max_fee, int) else None,
+            auto_estimate_fee=max_fee == "auto",
+            signer=signer,
+            account_address=account_address,
+            wait_for_acceptance=wait_for_acceptance,
+        )
+        self._logger.info(self.format_successful_invoke_response(response))
```

Comment:
what about failed result? we need to print out some info as well there

Reply:
Added a test case. `GatewayFacade` raises `ProtostarException`.